ui/apps/platform/src/Containers/Vulnerabilities/VirtualMachineCves/aggregateUtils.ts

```diff
@@ -198,3 +201,44 @@ export function applyVirtualMachineCveTableFilters(
         return true; // passed all filter conditions
     });
 }
+
+export function applyVirtualMachineCveTableSort(
+    rows: CveTableRow[],
+    sortKey: string,
+    reversed: boolean
+): CveTableRow[] {
+    const comparator = (a: CveTableRow, b: CveTableRow) => {
+        let compareResult = 0;
+
+        switch (sortKey) {
+            case CVE_SORT_FIELD:
+                // Intl.Collator sorting would be better here, but it's not used in the backend
+                compareResult = a.cve.localeCompare(b.cve);
+                break;
+            case CVSS_SORT_FIELD:
+                compareResult = a.cvss - b.cvss;
+                break;
+            case CVE_EPSS_PROBABILITY_SORT_FIELD:
+                compareResult = a.epssProbability - b.epssProbability;
+                break;
+            case CVE_STATUS_SORT_FIELD:
+                compareResult = Number(a.isFixable) - Number(b.isFixable);
+                break;
+            case CVE_SEVERITY_SORT_FIELD:
+                compareResult = severityRankings[a.severity] - severityRankings[b.severity];
+                break;
+            default:
+                break;
+        }
+        if (compareResult !== 0) {
+            return reversed ? compareResult * -1 : compareResult;
+        }
+
+        // backup compare when rows are equal
+        // doesn't appear to be a consistent behavior in the backend between vulnerability pages
+        // however secondary sort of cve name seems to mimic node cve page behavior
+        return a.cve.localeCompare(b.cve);
```

Comment:
Thinking out loud about lack of `reversed` that tie implies CVE name is not the sort field, so ascending order is intuitive for implicit secondary sort field.